tests/test_tex.py

```diff
@@ -60,6 +63,58 @@ def testSelectBinFromCorr(self):
         self.assertFloatsAlmostEqual(exp_rge1_avg_xip, obs_rge1_avg_xip, rtol=1e-7)
         self.assertFloatsAlmostEqual(exp_rge1_avg_xip_err, obs_rge1_avg_xip_err, rtol=1e-7)
 
+    def testEllipticityResidualZeroCorr(self):
+        """Does the correlation function correctly compute for a given ra, dec, e1, e2?
+
+        Our goal is mostly to check that we're calling this function correctly.
+        Tests of the deeper performance are implicitly outsourced to TreeCorr."""
+        nbins, log10_min_sep, log10_max_sep = 20, -0.55448274, 1.25345275
+        rand_seed = 1238625876
+        random.seed(rand_seed)
+        N = 100
+        ra = random.randn(N) * 20
+        dec = random.randn(N) * 20
+        g1 = random.randn(N) * 1e-1
+        g2 = random.randn(N) * 1e-1
+
+        exp_r = np.logspace(log10_min_sep, log10_max_sep, nbins) * u.arcmin
+        # Don't use zeros_like(exp_r) because that will include the units of exp_r
+        exp_xip = np.zeros(len(exp_r))
+        exp_xip_err = np.zeros(len(exp_r))
+
+        obs_r, obs_xip, obs_xip_err = correlation_function_ellipticity(ra, dec, g1, g2)
+        print("EXP_xip_err: ", repr(exp_xip_err))
+        print("OBS_xip_err: ", repr(obs_xip_err))
+        self.assertFloatsAlmostEqual(exp_r, obs_r, atol=1e-4 * u.arcmin)
+        self.assertFloatsAlmostEqual(exp_xip, obs_xip)
```

Comment:
If I'm reading this correctly, you're expecting the correlation function to be identically 0 everywhere?